docs/src/main/asciidoc/mongodb-panache-guide.adoc

```diff
@@ -0,0 +1,505 @@
+
+Clone the Git repository: `git clone {quickstarts-clone-url}`, or download an {quickstarts-archive-url}[archive].
+
+The solution is located in the `using-mongodb-panache` {quickstarts-tree-url}/using-mongodb-panache[directory].
+
+== Creating the Maven project
+
+First, we need a new project. Create a new project with the following command:
+
+[source,shell,subs=attributes+]
+----
+mvn io.quarkus:quarkus-maven-plugin:{quarkus-version}:create \
+    -DprojectGroupId=org.acme \
+    -DprojectArtifactId=using-mongodb-panache \
+    -DclassName="org.acme.rest.json.FruitResource" \
+    -Dpath="/fruits" \
+    -Dextensions="resteasy-jsonb,mongodb-panache"
+----
+
+This command generates a Maven structure importing the RESTEasy/JAX-RS, JSON-B and MongoDB Panache extensions.
+After this, the quarkus-mongodb-panache extension has been added to your `pom.xml`.
+
+If you don't want to generate a new project you can add the dependency in your `pom.xml`:
+
+[source,xml]
+--
+<dependencies>
+    <dependency>
+        <groupId>io.quarkus</groupId>
+        <artifactId>quarkus-mongodb-panache</artifactId>
+    </dependency>
+</dependencies>
+--
+
+== Setting up and configuring MongoDB with Panache
+
+To get started:
+
+* add your settings in `{config-file}`
+* Make your entities extend `PanacheMongoEntity`, you can use the `@MongoEntity` annotation to specify the name of the database and the name of the collection (it will default to the name of your entity).
+* place your entity logic in static methods in your entities
+
+Follow the link:mongo-guide.html#configuring-the-mongodb-database[Configuring the MongoDB database].
+
+Then add the relevant configuration properties in `{config-file}`.
+
+[source,properties]
+--
+# configure the mongoDB client for a replica set of two nodes
+quarkus.mongodb.connection-string = mongodb://mongo1:27017,mongo2:27017
+# mandatory if you don't specify the name of the database using @MongoEntity
+quarkus.mongodb.database = person
+--
+
+The `quarkus.mongodb.databas` property will be used by MongoDB Panache to detemine the name of the database where your entities will be persisted.
+
+== Defining your entity
+
+To define a Panache entity, simply extend `PanacheMongoEntity`, annotate it with `@MongoEntity` and add your
+columns as public fields:
+
+[source,java]
+--
+@MongoEntity(collection="ThePerson")
+public class Person extends PanacheMongoEntity {
+    public String name;
+    public LocalDate birth;
+    public Status status;
+}
+--
+
+NOTE: annotating with `@MongoEntity` is optional, it allows to configure the name of the collection and the name of the database. Here the entity will be stored in the `ThePerson` collection instead of the default `Person` collection.
+
+MongoDB with Panache uses the link:https://mongodb.github.io/mongo-java-driver/3.10/bson/pojos/[PojoCodecProvider] to map your entities to MongoDB Document. 
+
+You will be allow to use the following annotations to customize this mapping:
```

Comment:
Maybe we can rephrase this to:

1. MongoDB Panache allows the use the following annotations to customize this mapping..
*or*
2. You will be allowed to use the following annotations to customize this mapping

Reply:
second one done